enclave-runtime/src/lib.rs

```diff
@@ -368,14 +371,23 @@ pub unsafe extern "C" fn sync_parentchain(
 fn dispatch_parentchain_blocks_for_import<WorkerModeProvider: ProvideWorkerMode>(
 	blocks_to_sync: Vec<SignedBlock>,
 ) -> Result<()> {
-	match WorkerModeProvider::worker_mode() {
-		WorkerMode::Sidechain => GLOBAL_TRIGGERED_PARENTCHAIN_IMPORT_DISPATCHER_COMPONENT
-			.get()?
-			.dispatch_import(blocks_to_sync)?,
-		_ => GLOBAL_IMMEDIATE_PARENTCHAIN_IMPORT_DISPATCHER_COMPONENT
-			.get()?
-			.dispatch_import(blocks_to_sync)?,
-	}
+	let import_dispatcher =
+		if let Ok(solochain_handler) = GLOBAL_FULL_SOLOCHAIN_HANDLER_COMPONENT.get() {
+			solochain_handler.import_dispatcher.clone()
+		} else if let Ok(parachain_handler) = GLOBAL_FULL_PARACHAIN_HANDLER_COMPONENT.get() {
+			parachain_handler.import_dispatcher.clone()
+		} else {
+			return Err(Error::NoParentchainAssigned)
+		};
+
+	match import_dispatcher {
+		Some(dispatcher) => dispatcher.dispatch_import(blocks_to_sync)?,
+		None => match WorkerModeProvider::worker_mode() {
+			WorkerMode::Teeracle => trace!("Not importing any parentchain blocks"),
+			_ => return Err(Error::CouldNotDispatchBlockImport),
+		},
```

Comment:
How can the import dispatcher be none here?

Reply:
If a parentchain is not correctly initialized, it will be none.

Reply:
Actually, after reading your question, I guess it is very confusing. Let me fix this.

Reply:
Fixed in 1f93384a5c28a9483ff466f93ecad6e93ee23f81